chia/wallet/wallet_puzzle_store.py

```diff
@@ -151,17 +151,23 @@ async def get_derivation_record_for_puzzle_hash(self, puzzle_hash: str) -> Optio
 
         return None
 
-    async def set_used_up_to(self, index: uint32) -> None:
+    async def set_used_up_to(self, index: uint32, in_transaction=False) -> None:
         """
         Sets a derivation path to used so we don't use it again.
         """
-        async with self.db_wrapper.lock:
+
+        if not in_transaction:
+            await self.db_wrapper.lock.acquire()
+        try:
             cursor = await self.db_connection.execute(
                 "UPDATE derivation_paths SET used=1 WHERE derivation_index<=?",
                 (index,),
             )
             await cursor.close()
-            await self.db_connection.commit()
+        finally:
+            if not in_transaction:
+                await self.db_connection.commit()
```

Comment:
we don't want to commit this if we're in a transaction?

Reply:
is it because the transaction logic itself will commit this DB connection once it completes?

Reply:
yeah exactly.